ci/osx-deps

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -euo pipefail
+
+fold_start() {
+    key=$1
+    title=$2
+    echo -e "travis_fold:start:$key\e[2K"
+    echo -e "travis_time:start:$key\e[2K"
+    tick="$(date +%s)"
+    echo "$title"
+}
+
+fold_end() {
+    key=$1
+    tock="$(date +%s)"
+    nano=000000000
+    echo -e "travis_time:end:$key:start=$tick$nano,finish=$tock$nano,duration=$((tock - tick))$nano\e[2K"
+    echo -e "travis_fold:end:$key\e[2K"
+}
+
+fold_start Python "Install Python 3.8 from python.org"
+curl -O https://www.python.org/ftp/python/3.8.5/python-3.8.5-macosx10.9.pkg
+sudo installer -package python-3.8.5-macosx10.9.pkg -target /
+sudo ln -s /usr/local/bin/python3 /usr/local/bin/python
+hash -r
+fold_end Python
+
+fold_start ccache 'Install ccache (compile it ourselves)'
+curl -O -L https://github.com/ccache/ccache/releases/download/v3.7.11/ccache-3.7.11.tar.xz
+tar xf ccache-3.7.11.tar.xz
+pushd ccache-3.7.11
+./configure --prefix=/usr/local
+make
+make install
+popd
+for compiler in clang clang++ cc gcc c++ g++; do
+  ln -sf ccache /usr/local/bin/$compiler
+done
+fold_end ccache
+
+fold_start pkg-config 'Install pkg-config (compile it ourselves)'
+curl -O -L https://dl.bintray.com/homebrew/mirror/pkg-config-0.29.2.tar.gz
+tar xf pkg-config-0.29.2.tar.gz
+pushd pkg-config-0.29.2
+./configure --disable-debug --prefix=/usr/local --disable-host-tool \
+            --with-internal-glib --with-pc-path=/usr/local/lib/pkgconfig:/usr/local/share/pkgconfig:/usr/local/pkgconfig \
+            --with-system-include-path=/usr/include
+make
+make install
+popd
+fold_end pkg-config
+
+fold_start libpng 'Install libpng (compile it ourselves)'
+curl -O -L https://downloads.sourceforge.net/libpng/libpng-1.6.37.tar.xz
```

Comment:
We switched to using libpng from PIL...why do we need to install it manually?

Reply:
Let's see if we can do without it...

Reply:
Yes, that [build](https://travis-ci.com/github/matplotlib/matplotlib/jobs/367596267) seems to be working. I added libpng because I thought I had to compile freetype, but it seems that matplotlib manages to build freetype without it. This is good news, because we can drop also pkg-config, and then the dependency stage will take much less time.